src/ferc_xbrl_extractor/datapackage.py

```diff
@@ -423,14 +454,27 @@ def from_taxonomy(
             db_uri: Path to database required for a Frictionless resource.
             form_number: FERC form number used for datapackage name.
         """
-        resources = []
-        for role in taxonomy.roles:
-            for period_type in ["duration", "instant"]:
-                resource = Resource.from_link_role(role, period_type, db_uri)
-                if resource:
-                    resources.append(resource)
+        resources = {}
+        logger.info("Attempting to merge taxonomies into a single datapackage.")
+        for i, (taxonomy_version, taxonomy) in enumerate(sorted(taxonomies.items())):
```

Comment:
Merging taxonomies is a bit weird, so I'll try to explain the logic a bit. `Resource`s in a datapackage correspond to a single table, so this will parse a single taxonomy normally, then for all subsequent taxonomies, it will look for any new or missing tables. For tables that existed in the previous version and the new version, it will call `merge_resources`, which then looks at the columns as well as the primary key of the table.